lib/versions.js

```diff
@@ -0,0 +1,81 @@
+const chalk = require("chalk");
+
+/**
+ * Prints the versions overview
+ */
+class Versions
+{
+    /**
+     *
+     * @param {Object.<string,string>} packages
+     */
+    constructor (packages)
+    {
+        /**
+         * Mapping of package -> color to print the version number
+         * @type {Object<string, string>}
+         */
+        this.packages = packages;
+    }
+
+
+    /**
+     * Prints the version overview
+     */
+    print ()
+    {
+        const maxLength = this.getMaxPackageNameLength();
+
+        for (const packageName in this.packages)
+        {
+            if (!this.packages.hasOwnProperty(packageName))
+            {
+                continue;
+            }
+
+            const color = this.packages[packageName];
+
+            console.log(
+                chalk[color](packageName),
+                this.padding(packageName, maxLength),
+                require(`${packageName}/package.json`).version
+            );
+        }
+    }
+
+
+    /**
+     * Returns the max package name length
+     *
+     * @private
+     * @return {number}
+     */
+    getMaxPackageNameLength ()
+    {
+        return Object
+            .keys(this.packages)
+            .reduce((max, name) => Math.max(max, name.length), 0);
+    }
+
+
+    /**
+     * Returns the padding for the given
+     *
+     * @private
+     *
+     * @param {string} packageName
+     * @param {number} maxLength
+     * @return {string}
+     */
+    padding (packageName, maxLength)
+    {
+        const length = packageName.length;
+
+        return (length < maxLength)
+            ? " ".repeat(maxLength - length)
+            : "";
+    }
```

Comment:
Why no leftPad?

![](https://media2.giphy.com/media/YD94dyHmWRW0w/giphy.gif)

Reply:
![this-is-not-fine-001-dae9d5](https://user-images.githubusercontent.com/1032411/33985338-93d13568-e0ba-11e7-8ce0-6d2cc9bbf689.png)